src/main/java/com/epam/java/pdp/figures/Shape.java

```diff
@@ -44,24 +44,31 @@ void calculateLineLengths() {
 
     }
 
-    void arePointsOnTheSingleLine() {
+    boolean arePointsOnTheSingleLine() {
         calculateLineLengths();
         double maxLength = 0;
-//        for (int i=0; i< points.size(); i++){
-//            if (lines.get(i) > maxLength){
-//                maxLength = lines.get(i);
-//            }
-//        }
-        for (int i = 0; i < points.size(); i++) {
-//            int nextIndex = i + 1;
-            if (maxLength == lineLengths.get(i) + lineLengths.get(i++)) {
+
+        for (int i = 0; i < lineLengths.size(); i++) {
+            if (lineLengths.get(i) > maxLength) {
+                maxLength = lineLengths.get(i);
+            }
+        }
+        for (int i = 0; i < lineLengths.size(); i++) {
+            int nextIndex = i + 1;
+            if (nextIndex == lineLengths.size()) {
+                nextIndex = 0;
+            }
+            if (maxLength == lineLengths.get(i) + lineLengths.get(nextIndex)) {
```

Comment:
Use case 1) maxLength = 5 and the lineLengths collection is [3, 5, 2]? Then you receive message that "Points are not on the single line." which is wrong. And oneLine would be true, which is correct
Use case 2) maxLength = 5 and the lineLengths collection is [3, 2, 5]? Then you receive message that "Points are not on the single line." which is wrong. And oneLine would be false, which is wrong